src/main/java/com/localblox/ashfaq/filewatcher/HdfsFileWatcher.java

```diff
@@ -0,0 +1,172 @@
+ * file.
+ *
+ * <p/>
+ * It was agreed that uncomplete file will have '.' (dot) before the name. Basic file handling flow should be as
+ * follow:
+ *
+ * <br/> 1. Create file in folder IN or OUT prepended with dot. for example 'in/.aaaaaa-bbbb-1234-5678.csv'
+ *
+ * <br/> 2. Write content to the file 'in/.aaaaaa-bbbb-1234-5678.csv'
+ *
+ * <br/> 3. rename file 'in/.aaaaaa-bbbb-1234-5678.csv' to 'in/aaaaaa-bbbb-1234-5678.csv' that it will match with
+ * pattern and will be processed.
+ */
+public class HdfsFileWatcher {
+
+    private static final Logger log = LoggerFactory.getLogger(HdfsFileWatcher.class);
+
+    private String hdfsAdminUri;
+
+    // TODO check pattern for GIUD
+    // default file name as GUID
+    private String inFilePattern = "in/[a-fA-F0-9\\-].csv";
+
+    // TODO check pattern for GIUD
+    // default file name as GUID
+    private String outFilePattern = "out/[a-fA-F0-9\\-].csv";
+
+    /**
+     * Creates HdfsFileWatcher instance.
+     *
+     * @param hdfsAdminUri - HDFS admin URI
+     */
+    public HdfsFileWatcher(final String hdfsAdminUri) {
+        this.hdfsAdminUri = hdfsAdminUri;
+    }
+
+    /**
+     * Creates HdfsFileWatcher instance.
+     *
+     * @param hdfsAdminUri   - HDFS admin URI
+     * @param inFilePattern  - input file pattern for ready to process files
+     * @param outFilePattern - output file pattern for ready to process file
+     */
+    public HdfsFileWatcher(final String hdfsAdminUri, final String inFilePattern, final String outFilePattern) {
+        this.hdfsAdminUri = hdfsAdminUri;
+        this.inFilePattern = inFilePattern;
+        this.outFilePattern = outFilePattern;
+    }
+
+    // TODO check if processing will be in multiple thread. If so - use AtomicBoolean
+    private boolean proceed;
+
+    /**
+     * Start the watcher to process file.
+     *
+     * This method will block until processing is stopped.
+     */
+    //TODO - think about execution context inside Spark and blocking. May be run in separate thread if need.
+    //TODO - if there should be more than one file watcher - resolve multiple file processing issue (one file
+    // processed twice).
+    public void start() {
+
+        proceed = true;
+
+        DFSInotifyEventInputStream eventStream = getDfsInotifyEventInputStream();
+        while (proceed) {
+            EventBatch events = null;
+            log.info("proceed event listening: {}", proceed);
+            try {
+                // TODO - move to configuration.
+                events = eventStream.poll(5L, TimeUnit.SECONDS);
+
+                // TODO this is draft logic, need to be reviewed according to requirements
+                if (events != null) {
+                    for (Event event : events.getEvents()) {
+                        log.info("event type: {}", event.getEventType());
+                        switch (event.getEventType()) {
+                            case RENAME:
```

Comment:
@gutsal-arsen 
OK, lets try to go in this way. 
we need to check real event behavior because i have thought that close event can be fired twice 
because of this javadoc: 
/**
   * Sent when a file is closed after append or create.
   */
  public static class CloseEvent extends Event {